src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java

```diff
@@ -58,37 +61,37 @@
 
 	private ZMQ.Socket subscriber;
 	private ZMQ.Context context;
+	private ObjectMapper mapper = new ObjectMapper();
 
 	{
 		context = ZMQ.context(1);
 	}
 
 	public void receive() {
 		getSubscriber();
-		String exportString = null;
-		byte[] exportBytes = null;
-		Event event;
+		JsonNode node;
+		String msgEnvenlope = null;
 		logger.info("Watching for new exported Event messages...");
 		try {
 			while (!Thread.currentThread().isInterrupted()) {
-				if (exportClient) {
-					if (serializedJava) { // supporting legacy Java export distro that shipped Serialized eventString
-						exportString = subscriber.recvStr();
-						event = toEvent(exportString);
-					} else { // supporting new Go export distro that ships JSON
-						exportBytes = subscriber.recv();
-						event = toEvent(exportBytes);
+				msgEnvenlope = subscriber.recvStr();
+				if (!"events".equals(msgEnvenlope)) { // why is every other event "events"?
+					node = mapper.readTree(msgEnvenlope);
+					switch (payloadType(node)) {
+					case NO_ENVELOPE:
+						processEvent(node);
+						break;
+					case JSON:
+						processJsonEvent(node);
+						break;
+					case CBOR:
+						processCborEvent(node);
+					default:
+						logger.error("Unknown payload type received");
+						break;
 					}
-				} else {
-					exportBytes = subscriber.recv();
-					event = toEvent(exportBytes);
-				}
-				if (event == null) {
-					logger.info("CBOR is unsupported in the rules engine at this time; message being ignored");
-				} else {
-					engine.execute(event);
-					logger.info("Event sent to rules engine for device id:  " + event.getDevice());
-				}
+				} else
+					logger.debug("Received 'events' string");
```

Comment:
Remove now that this is explained.

Reply:
yep - gone along with the entire conditional now that I am addressing multipart correctly.